### ISSUE TITLE
Design Architecture

### ISSUE THREAD

**Issue by Jen-Lopez** [Jen-Lopez/movie-review-app#7]
How are we going to build the API?
How are API tokens stored/accessed on android? 
How would server handle API Calls?

**Acceptance Criteria:**
Create an Architecture Diagram

**Comment by lucasortizny** [comment]
In order to make any API calls to TMDB, there would be no safe way to make the calls if we are using our own API keys. So we should come up with a way to authenticate users + any calls they make from the app will first go through our own servers where WE make the calls server-side and return the results. That way, our TMDB API Key is not exposed in the APK during packaging. Here are my ideas:

1) Use the Google API to authenticate users into our app (and we can grab their emails and stuff that way so we don't have to store sensitive information).
2) Verify if the user is logged in by passing in their account ID on their API calls to our server. That account ID is gonna be the primary key for the data hosted in our databases for identifying the user. 
3) Once that user is authenticated and verified, any API calls to our server should return the data by making an API call server-side for this. 

Will look into finding ways to refine this but until then this is what my idea should be. 
TLDR = move any TMDB calls to the server side, only keep API key on a per-user basis for the Google API to authenticate on OUR own servers (I will draw a map later of what I have in mind).

**Comment by Jen-Lopez** [comment]
We can build the REST API with Node.JS/Express and connect it to the android client using libraries like retrofit. 

**Resources**
- [REST API](https://www.youtube.com/watch?v=pKd0Rpw7O48)
- [Retrofit Library](https://medium.com/android-news/consuming-rest-api-using-retrofit-library-in-android-ed47aef01ecb#:~:text=Retrofit%20is%20a%20REST%20Client,for%20example%20SimpleXML%20and%20Jackson.)
